project-koku/koku
# [COST-348] Update provider_db_accessor to not return first provider if no uuid is passed in

Previously, when no provider uuid or auth_id were supplied, it would grab the first provider in the table. This change makes it to where if both are not supplied, it returns None instead of the first provider.

To Test:

1. Load in a little data: 
     -`make create-test-customer`
     -`make load-test-customer-data`
2. Launch the Django Shell:
     -`make shell`
2. Import the update_summary_tables function in the shell:  
     -`from masu.processor.tasks import update_summary_tables`
3. Try to call update_summary_tables in the shell with a uuid of None and you should see an Error that the provider was not found, example call: 
     -`update_summary_tables("acct10001", "AWS-local", None, "2020-07-01")`
4. If you want to try calling it with a provider, go to the sources endpoint and replace None with the provider uuid and see that it starts updating properly, example call: 
     -`update_summary_tables("acct10001", "AWS-local", {uuid}, "2020-07-01")`

Smokes Results: `3947 passed, 80 skipped, 1709 deselected in 898.39s (0:14:58)`
